tool/tsh/common/tsh.go

```diff
@@ -1815,6 +1815,12 @@ func onLogin(cf *CLIConf) error {
 		}
 	}
 
+	// If the cluster is using single-sign on, providing the user name
+	// with --user is likely a mistake, so display a warning.
+	if cf.AuthConnector != "" && cf.AuthConnector != constants.LocalConnector && cf.Username != "" {
```

Comment:
what about `passwordless` connector? At least for dev setup I am using user for passwordless login because it skips one dialog when multiple credentials are stored on authenticator. 
wdyt @codingllama ?